pkg/types/baremetal/validation/platform.go

```diff
@@ -404,23 +405,27 @@ func ValidatePlatform(p *baremetal.Platform, n *types.Networking, fldPath *field
 		}
 	}
 
-	if p.Hosts == nil {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("hosts"), p.Hosts, "bare metal hosts are missing"))
+	pathName, _ := os.Executable()
+	if !strings.Contains(pathName, "agent") {
```

Comment:
This worked while debugging the test case but did not work when building the binary and running the create image command. Need to look more into it.